Lib/ldap/cidict.py

```diff
@@ -35,7 +35,7 @@ def __delitem__(self,key):
     del self.data[lower_key]
 
   def update(self,dict):
-    for key in dict.keys():
+    for key in dict:
```

Comment:
You can drop the whole function. ``UserDict.IterableUserDict`` and ``collections.Mapping`` always provide a correct ``update()``.

Reply:
Using the parent implementation results in not correctly transforming the keys to lowercase on Python 2.

See an experimental test case here:

https://travis-ci.org/jdufresne/python-ldap/builds/318122299